content/blog/en/version-6-dojo.md

```diff
@@ -0,0 +1,295 @@
+        <button
+          onclick={this._decrement}
+          classes={[css.button, css.decrement]}
+        />
+        <div classes={[css.counter]}>{`${this._count}`}</div>
+        <button
+          onclick={this._increment}
+          classes={[css.button, css.increment]}
+        />
+      </div>
+    );
+  }
+}
+```
+
+Using function-based widgets, we define `icache` middleware for the widget factory so that we can store the counter state across renders. Notice that when we call set on the `icache` we don't need to manually `invalidate` the widget as `icache` implicitly invalidates when a value is set.
+
+> src/FunctionalCounter.tsx
+```tsx
+interface FunctionalCounterProperties {
+  incrementStep?: number;
+}
+
+const factory = create({ icache }).properties<FunctionalCounterProperties>();
+
+export const FunctionalCounter = factory(function FunctionalCounter({
+  middleware: { icache },
+  properties
+}) {
+  const { incrementStep } = properties();
+  const count = icache.get<number>("count") || 0;
+  return (
+    <div classes={[css.root]}>
+      <button
+        onclick={() => {
+          icache.set("count", count - incrementStep);
+        }}
+        classes={[css.button, css.decrement]}
+      />
+      <div classes={[css.counter]}>{`${count}`}</div>
+      <button
+        onclick={() => {
+          icache.set("count", count + incrementStep);
+        }}
+        classes={[css.button, css.increment]}
+      />
+    </div>
+  );
+});
+```
+
+For more information please see the [middleware reference guides](/learn/middleware/introduction) or some of the middleware examples on codesandbox:
+
+ * [`theme` middleware](https://codesandbox.io/s/theme-middleware-4btv7)
+ * [`icache` middleware](https://codesandbox.io/s/advanced-icache-middleware-teeig)
+
+Don't worry, the existing class-based widget APIs are not going away! These enhancements are additive and backwards-compatible, providing what we believe to be an ergonomic improvement on top of the existing widget APIs.
+
+We are really looking forward to seeing the fun and innovative middlewares from the Dojo community, and, as always, please let us know any feedback that you might have!
+
+## Config free Custom Elements
+
+Including Web Components as a first class citizen in Dojo is something that we’ve always been passionate about and now compiling your Dojo widgets to Custom Elements is _even_ easier. In Dojo 6 there is no configuration required, other than defining the widget(s) in the `.dojorc` to instruct the `@dojo/cli-build-widget` to compile your widgets to custom elements. We think this is a significant improvement to current tooling that requires widgets to be explicitly configured with the custom element details. Doing so takes additional development effort, foresight, and creates an additional maintenance burden of keeping the configuration up to date with changes to widget properties. This can be seen below with the custom element configuration for an example widget.
```

Comment:
```suggestion
Including Web Components as a first class citizen in Dojo is something that we’ve always been passionate about and now compiling your Dojo widgets to Custom Elements is _even_ easier. In Dojo 6 there is no configuration required, other than defining the widget(s) in the `.dojorc` to instruct the `@dojo/cli-build-widget` to compile your widgets to custom elements. We think this is a significant improvement to current tooling that requires widgets to be explicitly configured with the custom element details. The approach prior to Dojo version 6 required additional development effort, foresight, and created an additional maintenance burden of keeping the configuration up to date with changes to widget properties. This can be seen below with the custom element configuration for an example widget.
```